docs/development/js-bundling.md

```diff
@@ -0,0 +1,177 @@
+# JS Bundling
+
+We are using [`rollup.js`](https://rollupjs.org/) for module-bundling. This process combines multiple JavaScript files
+and bundles them together into one, which is production-ready and loadable in the browser. This reduces network traffic
+while allowing us to manage our business logic in smaller portions.
+
+**Bundling a service, module or component is optional altough highly recommended.**
+
+## Bundle Structure
+
+If you want to bundle your business logic, please stick with the following file-structure:
+
+```
+.
+├── dist                contains bundled files for the browser.
+├── src                 contains the business logic.
+    └── index.js        index-file which gets bundled.
+└── rollup.config.js    config for the module-bundler.
+```
+
+As indicated by the structure above, the `index.js` file serves as an entry-point to the JavaScript bundle. It is also
+the file which ultimately gets bundled by our module-bundler. The name may be something different than `index`, but it
+should still indicate this intention.
+
+The configuration file `rollup.config.js` must be located in the bundle root-directory and its name must not be changed.
+How this file might look like will be covered in the next chapter.
+
+## Rollup Configuration
+
+This section will list a few common configurations used in the ILIAS JavaScript codebase. For more advanced use-cases
+you can have a look at the full documentation of rollup [here](https://rollupjs.org/configuration-options/).
```

Comment:
Please do not use “here” as link text. See e.g. the checkpoint about [meaningful link targets](http://www.w3.org/TR/2000/NOTE-WCAG10-TECHS-20001106/#tech-meaningful-links) in [Techniques for Web Content Accessibility Guidelines 1.0](http://www.w3.org/TR/2000/NOTE-WCAG10-TECHS-20001106/). I’d suggest something along the lines of

```gfm
you can have a look at the list of [configuration options](https://rollupjs.org/configuration-options/) in the Rollup documentation.
```

What do you think?